### ISSUE TITLE
Export a list of pictures names in a collection

### ISSUE THREAD

**Issue by aurelienpierre** [darktable-org/darktable#7563]
Sometimes, the person doing the culling in a photo collection is not the person doing the editing and might not even be the photographer either. Therefore, we need a quick way to communicate a list of chosen pictures between photographer, retoucher and artistic/creative director, maybe even with the customer when there is one.

## Proposal

1. A button somewhere in lighttable that exports the file names of the pictures in the current collection, in a text file, possibly with stars rating, color labels and tags, for example :
```text
collection.txt
$(BASE_DIRECTORY)/$(COLLECTION_FOLDER)/DSC_0005.NEF ; 4 stars ; red, blue ; tag1, tag2, tag3 ;
$(BASE_DIRECTORY)/$(COLLECTION_FOLDER)/DSC_00012.NEF ; 1 stars ; yellow ; tag1, tag2, tag3 ;
$(BASE_DIRECTORY)/$(COLLECTION_FOLDER)/DSC_00015.NEF ; 5 stars ; red, blue ; tag1, tag2 ;
etc.
```
where the file name of the export file (`collection.txt`) can be input by user and the `$(XXX)` masks are replaced with the actual folders. Stars ratings, tags and color labels could be optional.

2. A button somewhere in lighttable that imports the file names of the pictures from the exported text file to create a new collection and set it active in lighttable. The import will let user choose if the `$COLLECTION_FOLDER` is to be kept as-is or changed to account for the local repository structure, while the `$(BASE_DIRECTORY)` will be updated silently depending on the global parameter of dt or changed upon user request.

## Further extension

This could later be extended by optionally adding:
1. the XMP history file of each picture in the collection,
2. the RAW files,
3. the low-res JPEG previews

and zip them altogether.

## Implementation

I feel this could be implemented in core and written in C or implemented in an extension and written in Lua. I see advantages and drawbacks for both, I'm not sure what's best.

**Comment by johnny-bit** [comment]
Sooo... Something like contact sheet but in txt form?

**Comment by wpferguson** [comment]
I suggest do it in Lua, let it go through a couple of iterations, then move it to core

**Comment by chhil** [comment]
Why selection  from a collection ?
Use tags that can be filtered across collections for export with details. 
Maybe an import to a collection (which could be done by filtering on tag and then moving them to a collection if the grouping into collection 8s desired).

**Comment by aurelienpierre** [comment]
> Why selection from a collection ?

Because the usual case is this : you have done a photo session of hundreds, if not thousands, of pictures. In that session, you want to tell your retoucher to edit pictures number 5, 258 and 762. So the list you produce is a subset of a collection.

>  Use tags that can be filtered across collections for export with details.

No. Tags are for archival purposes, aka long-term use. Multiplying short-lived tags is not a solution, since it will pollute the database with convenience tags you will have to cleanup every once in a while. Also, the point is to share and communicate info to other persons, aka other darktable installations, so polluting their database with your convenience tags is not polite.

(I have a load of shit tags I gather from opening other people's XMP, some of them in languages I don't even speak, it's really annoying).

> Sooo... Something like contact sheet but in txt form?

Yes, kind of.

**Comment by wpferguson** [comment]
Started playing with it

![wfexport](https://user-images.githubusercontent.com/1634891/103395334-bd69b400-4afb-11eb-8fce-52c8514f2e67.jpg)

and

[collection.txt](https://github.com/darktable-org/darktable/files/5756475/collection.txt)

**Comment by wpferguson** [comment]
@aurelienpierre a question...

* You shoot a set of pictures.  
* You select the ones you want edited.  
* You send the list of images, with color codes and instructions, and optionally images and/or xmps, to the editor.  
* They do the edits. 

What gets returned to the shooter?  xmps or exported images or ?

**Comment by aurelienpierre** [comment]
Well, if everybody "speaks" darktable, exchanging only RAWs the first time, then XMPs for communication, is more bandwidth efficient. Of course, if one of the parties doesn't speak dt, only exported images and list of names can be exchanged.

**Comment by wpferguson** [comment]
Started working on this again because I had culled images in dt 3.6, but wanted to edit them in 3.7 with the new toys :smile:.  I already had the film roll imported in both, so ran the workflow collaboration script I had worked on earlier to generate the "manifest".  I then built the part to import the manifest, find the applicable film roll, and then apply the ratings.  So, progress.

One gotcha is that the sort order needs to be the same for the source and destination, otherwise settings didn't get applied correctly.
